docs/CreateRecord-transform.md

```diff
@@ -0,0 +1,62 @@
+# Create Record transform
+
+Description
+-----------
+Transform plugin generates hierarchical structures from flat schemas.
+
+Use Case
+--------
+Collapse flat data into structures.
+
+
+Properties
+----------
+**Mapping:** Specifies the mapping for generating the hierarchy.
+**Include fields missing in mapping:** Specifies whether the fields in the input schema that are not part of the mapping,
+should be carried over as-is.
+
+Example
+-------
+Let's say we have two following data structure:
+
+| field                 | type         |
+| ------------          | ------------ |
+| customer_id           | Int          |
+| customer_name         | String       |
+| customer_phone        | String       |
+| order_id              | Int          |
+| product_id            | Int          |
+| amount                | Double       |
+| order_description     | String       |
+
+By setting the following mapping:
+
+```
+{
+	"id": ["customer_id"],
+	"customer": {
+		"name": ["customer_name"],
+		"phone": ["customer_phone"]
+	},
+	"orders": {
+		"id": ["order_id"],
+		"product_id": ["product_id"],
+		"amount": ["amount"],
+		"description": ["order_description"]
```

Comment:
Please provide an example mapping for the case where the hierarchy is more than 1 level deep. Please add a unit test for this case as well.

Reply:
updated